### PR TITLE
feat: add Overview storage capacity summary

### LINKED ISSUE #158: [FEATURE] Add Overview disk capacity and DustFril-detected developer storage share
## Summary

Add a storage summary to the Desktop Overview that shows the capacity/usage of the filesystem volume containing the current workspace and explains how much of that used storage is represented by development data that DustFril has actually measured.

This feature must not claim to know the size of **all developer files on the machine** unless DustFril has actually measured that scope.

The intended user-facing concept is:

```text
Storage

512 GB total
318 GB used
194 GB available

DustFril-detected development storage
34 GB · 10.7% of used storage
```

The exact visual treatment may evolve, but the storage semantics must remain precise and Core-owned.

## Motivation

DustFril already measures scanner-owned development artifacts such as `target/`, `node_modules/`, and supported Java build outputs. The Overview currently shows cleanup-oriented metrics, but it does not give users the broader context:

```text
How full is the disk/volume containing this workspace?
How much storage is still available?
How large is the development storage DustFril has detected relative to used disk space?
```

This makes the Overview more useful without turning DustFril into a generic full-disk analyzer.

Related historical work:

- #15 implemented artifact disk-usage analysis, not whole-volume capacity reporting.
- #72/#73 track generated artifact sizes/history, not total disk capacity.
- #100 listed disk-usage visualization as a future extension.

## Scope / terminology

Use accurate terminology.

Do **not** label the metric simply:

```text
Developer files: 12%
```

unless DustFril has actually measured every developer-owned file represented by that claim.

Prefer wording equivalent to:

```text
Detected development storage
DustFril-detected storage
Development artifacts
```

with scope/context available where useful.

The numerator is an evidence-backed measured value, not an estimate based on file extensions or guessed folder names.

## Volume semantics

The volume being reported should be the filesystem volume that contains the current selected workspace.

This matters because the workspace may live on:

- the system disk;
- an external disk;
- another mounted volume.

Report structured values such as:

```text
total_bytes
used_bytes
available_bytes
```

Derive `used_bytes` consistently from the platform/filesystem values and document the exact formula.

Do not hard-code `/` or assume the main system volume if the selected workspace belongs to another mounted filesystem.

If there is no selected workspace, either:

1. show a neutral no-workspace state; or
2. use an already established default workspace/root only if that behavior is explicit in the current app.

Do not silently switch between unrelated volumes.

## Developer-storage numerator

For the initial implementation, count only storage DustFril has already measured through supported Core-owned analysis.

At minimum this may include current scanner-owned generated artifacts from the selected analyzed scope, for example:

```text
Rust target/
Node node_modules/
Java supported build output
```

As later features are implemented, the same model may be extended deliberately to include other measured developer-storage categories such as supported developer caches from #146.

Do not include new categories merely by path-name heuristics.

Do not recursively scan the full disk/home directory solely to populate this Overview metric.

## Share calculation

The primary percentage should be defined as:

```text
detected_developer_bytes / volume_used_bytes * 100
```

because the user question is how much of the **currently used storage** is represented by DustFril-detected development data.

Handle edge cases explicitly:

```text
used_bytes == 0
unknown/unavailable filesystem statistics
no completed workspace analysis
partial analysis
```

Do not divide by total capacity unless the UI labels that metric differently.

If both percentages are ever shown, distinguish them clearly:

```text
10.7% of used storage
6.6% of total capacity
```

The initial UI should prefer one clear percentage rather than unnecessary duplicates.

## Coverage / honesty

The Overview must make clear when the development-storage number reflects only the selected workspace/analyzed scope.

Example:

```text
34 GB detected in current workspace
10.7% of used storage on this volume
```

This is not the same as claiming:

```text
All developer files on this Mac = 34 GB
```

If future DustFril versions aggregate multiple registered workspaces/caches, add an explicit coverage model rather than silently changing the meaning of this metric.

A structured model may include something conceptually equivalent to:

```text
DeveloperStorageSummary
  measured_bytes
  scope
  categories
  analysis_timestamp
  partial/warnings
```

Exact names are flexible.

## Reuse existing analysis

This is a hard requirement.

Do not perform:

```text
workspace scan/analyze
-> Overview opens
-> recursively measure all artifacts again
```

Reuse the most recent valid `AnalysisResult`, snapshot, or another existing Core-owned measured result where semantics match.

If no valid analysis exists for the selected workspace, the Overview should show a state such as:

```text
Not analyzed yet
```

and let the normal Workspace analysis produce the data.

Opening or refreshing Overview must not create another expensive recursive scan merely to calculate the percentage.

## Core architecture

Core owns storage semantics.

Preferred dependency direction:

```text
Desktop Overview
      ↓
Tauri command / DTO
      ↓
dustfril-core storage summary API
      ├ volume filesystem statistics
      └ existing measured development-artifact data
```

Do not calculate volume usage or developer-storage semantics independently in React.

The frontend may format raw bytes and percentages for display.

## Cross-platform behavior

Use platform/filesystem APIs or a small well-supported Rust abstraction that can report volume capacity for the workspace path.

Validate at least the platforms currently claimed by DustFril.

Requirements:

- no shell command parsing such as invoking `df` just to get storage statistics when a Rust/system API is available;
- workspace path resolves to the correct containing volume;
- filesystem errors are reported explicitly;
- unsupported platform behavior is honest rather than fabricated.

Do not add broad platform-specific infrastructure beyond what is required for this metric.

## Overview UX direction

Keep the Overview compact and Finder-like rather than turning it into a dashboard full of large cards.

A possible presentation:

```text
STORAGE

318 GB used of 512 GB                         194 GB available
████████████████████░░░░░░░░

Development artifacts     34 GB · 10.7% of used storage
```

or an equivalent compact layout.

The visualization must remain understandable without relying on color alone.

Avoid large decorative charts unless they add clear value.

Do not display excessive decimal precision.

## Relationship to cleanup metrics

Keep these concepts separate:

```text
Detected development storage
    -> measured supported development data in the analyzed scope

Recommended/reclaimable storage
    -> subset DustFril currently recommends for cleanup

Selected storage
    -> exact user-selected cleanup candidates
```

Do not present all detected developer storage as reclaimable.

For example:

```text
Detected development storage: 34 GB
Recommended cleanup: 8 GB
```

can both be true.

## Performance requirements

Fetching filesystem capacity should be cheap and must not trigger a full tree walk.

The developer-storage portion should reuse existing measured data.

Opening Overview repeatedly must not:

- rescan the workspace;
- recalculate every artifact directory size;
- create Activity History entries;
- create artifact snapshots;
- mutate cleanup recommendations.

## Tests

Add deterministic tests covering at minimum:

### Volume metrics

- total/used/available relationship is internally consistent;
- workspace path maps to its containing filesystem/volume;
- invalid/missing path is handled explicitly;
- filesystem-stat failure is not rendered as zero capacity;
- percentage calculation handles zero/unknown used bytes safely.

### Developer storage

- existing analysis total is reused;
- no analysis -> explicit unavailable/not-analyzed state;
- mixed Rust/Node/Java analyzed artifacts aggregate correctly;
- detected total includes Keep/NeedsReview/Recommended artifacts because it represents measured development storage, not only cleanup recommendations;
- reclaimable/recommended bytes remain a separate metric;
- nested/covered artifacts do not double-count;
- opening Overview does not trigger another recursive analysis.

### Scope honesty

- selected-workspace scope is represented explicitly;
- a workspace on another volume uses that volume's capacity rather than the system disk;
- percentage is calculated against used bytes according to the documented formula.

## Manual validation

In the real Tauri app:

1. select a workspace on the normal development volume;
2. analyze it;
3. open Overview;
4. verify total/used/available values against the operating system's storage information within expected filesystem-reporting differences;
5. verify detected development bytes match the current analyzed artifact totals;
6. verify percentage math;
7. switch to another workspace and confirm stale values are not retained;
8. if available, select a workspace on another mounted volume and verify the volume statistics change accordingly;
9. reopen Overview repeatedly and confirm no new scan/history/snapshot operation is created.

## Quality gates

Run the repository's maintained checks, at minimum:

```bash
cargo fmt --all -- --check
cargo clippy --workspace --all-targets -- -D warnings
cargo test --workspace
cargo llvm-cov --workspace --all-features --summary-only
git diff --check
```

Run maintained frontend checks/build and real Tauri smoke testing for Overview.

## Non-goals

- recursively measuring the entire disk;
- full-home filesystem indexing;
- claiming to identify all developer files on the machine;
- duplicate-file analysis;
- large-file finder;
- arbitrary user-file categorization;
- deleting files from Overview;
- background storage monitoring daemon;
- real-time OS filesystem watcher;
- cloud storage reporting;
- per-folder treemap/disk-visualizer implementation;
- new cleanup artifact heuristics solely to increase the percentage.

## Acceptance criteria

- [ ] Overview reports total, used, and available bytes for the volume containing the selected workspace.
- [ ] Developer-storage bytes come only from existing DustFril-measured Core data.
- [ ] The percentage is defined as detected developer bytes / used volume bytes and is tested.
- [ ] UI wording does not imply DustFril measured all developer files on the machine.
- [ ] Detected storage remains distinct from recommended/reclaimable storage.
- [ ] Keep/NeedsReview/Recommended artifacts all contribute to detected development storage when present in the valid analyzed result.
- [ ] Overview reuses existing measured results and does not recursively rescan solely for display.
- [ ] Workspaces on different mounted volumes report the correct containing volume.
- [ ] Unsupported/error/not-analyzed states are explicit rather than misleading zeroes.
- [ ] Core owns the semantics; Tauri/React remain presentation/integration layers.
- [ ] Real Tauri Overview smoke testing passes without creating duplicate scan/history/snapshot operations.

This feature should remain independent of the current v0.0.1 release gate unless the release plan is explicitly changed.

### PR DESCRIPTION
**Pull request by mors119**
## What

Closes #158

- Add a Core-owned storage summary API backed by filesystem statistics for the volume containing the selected workspace.
- Reuse the completed `AnalysisResult` to report measured scanner-owned development storage, its share of used volume bytes, and separate recommended cleanup bytes.
- Propagate analyzer measurement failures so partial storage results are not presented as complete.
- Add explicit not-analyzed, unavailable, and partial-analysis states with scope context.
- Render a compact Finder-like Storage section in Desktop Overview without triggering another workspace scan.
- Refresh volume capacity after permanent cleanup without re-scanning or re-analyzing the workspace.

## Why

Overview now answers how full the selected workspace’s volume is while keeping the measured development-storage claim limited to DustFril-supported, analyzed artifacts. Keep, Needs Review, and Recommended artifacts contribute to detected storage; only Recommended bytes are shown as cleanup candidates.

## Validation

- `cargo fmt --all -- --check` ✅
- `cargo clippy --workspace --all-targets -- -D warnings` ✅
- `cargo test -p dustfril-core --lib -- --skip signed_system_fixture_is_reported_as_valid` ✅ (301 tests)
- `cargo test -p dustfril-tauri --lib` ✅ (19 tests)
- `npm test -- --run` ✅ (28 tests)
- `npm run build` ✅
- `git diff --check` ✅
- GitHub Rust CI ✅
- GitHub Tauri Check ✅

`cargo test --workspace` was also run. It has one existing macOS signing fixture assertion failure in `crates/core/src/integrity/macos.rs` (`macOS Software Signing` vs `Software Signing`); no unrelated files were changed.

### CODE REVIEW

File: apps/tauri/src/features/app-shell/hooks/useAppState.ts [hunk 1]
```diff
@@ -319,6 +323,34 @@ export function useAppState() {
...
+              ...current,
+              detectedDevelopmentBytes: Math.max(
+                0,
+                current.detectedDevelopmentBytes - deletedDetectedBytes,
+              ),
+              detectedSharePercent:
+                current.usedBytes > 0
+                  ? (Math.max(0, current.detectedDevelopmentBytes - deletedDetectedBytes) /
+                      current.usedBytes) *
+                    100
```
[the remark this answers is not in the document]
Comment (mors119) [reply apps/tauri/src/features/app-shell/hooks/useAppState.ts]: Adjust로 반영했습니다 (79944d7). 영구 cleanup에서 실제 삭제가 발생한 경우에만 Core의 volume()을 통해 Tauri refresh_storage_volume을 호출하고, re-scan/re-analysis 없이 total/used/available 및 share를 갱신합니다. Trash에는 호출하지 않으며, refresh 실패 시 cleanup 성공은 유지하고 warning만 표시합니다. Tauri contract와 frontend 회귀 테스트를 추가했습니다.

File: crates/core/src/storage.rs [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+    let mut categories = analysis
+        .artifacts
+        .iter()
+        .map(|artifact| artifact.artifact.ecosystem)
+        .collect::<Vec<_>>();
+    categories.sort_unstable();
+    categories.dedup();
+
+    let partial = access_summary.is_some_and(|summary| summary.failures > 0);
```
[the remark this answers is not in the document]
Comment (mors119) [reply crates/core/src/storage.rs]: 반영했습니다 (79944d7). Analyzer가 WalkDir, metadata, modified time 측정 실패를 AnalysisResult.measurement_failures로 전달하고, storage summary가 scan access failure와 함께 partial/warning으로 표시하도록 수정했습니다. 측정된 바이트는 그대로 유지하며, 실패가 0인 기존 wire payload는 변경되지 않습니다. 관련 analyzer/storage 회귀 테스트도 추가했습니다.
